server/test/unit/converters/schema/test_gene_symbol.py

```diff
@@ -0,0 +1,45 @@
+import os
+import unittest
+
+import pandas as pd
+
+from server.test import FIXTURES_ROOT
+from server.converters.schema import gene_symbol
+
+
+class TestHGNCSymbolChecker(unittest.TestCase):
+
+    def setUp(self):
+        self.test_hgnc_path = os.path.join(FIXTURES_ROOT, "hgnc_example.txt.gz")
+        self.hgnc_checker = gene_symbol.HGNCSymbolChecker.from_hgnc_records(self.test_hgnc_path)
+
+    def test_symbol_upgrade(self):
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("SEPT1"), "SEPTIN1")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("ADRB2R"), "ADRB2")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("BAR"), "ADRB2")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("sept1"), "SEPTIN1")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("AdRb2R"), "ADRB2")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("bar"), "ADRB2")
+
+        # DIFF6 is ambiguous so don't upgrade it
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("DIFF6"), "DIFF6")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("diff6"), "diff6")
+
+        # ARG1 is approved
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("ARG1"), "ARG1")
+        self.assertEqual(self.hgnc_checker.upgrade_symbol("arg1"), "ARG1")
+
```

Comment:
Any symbols that are withdrawn and approved that should check out to be approved?

Reply:
Good idea, added a test for that.